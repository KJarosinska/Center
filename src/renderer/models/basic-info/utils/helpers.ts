import { SimCard } from "Renderer/models/basic-info/interfaces"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter(simCard => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
