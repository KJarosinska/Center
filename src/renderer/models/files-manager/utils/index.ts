import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
