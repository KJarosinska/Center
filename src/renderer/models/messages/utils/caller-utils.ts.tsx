import { Author } from "Renderer/models/messages/messages.interface"

const removeDecoratorsFromPhoneNumber = (string = ""): string => {
  return string.split(" ").join("").replace("+", "")
}

export interface CallerSearchParams extends Record<string, string> {
  id: string
  phoneNumber: string
}

export const isCallerMatchingSearchParams = (
  caller: Author,
  params: CallerSearchParams
): boolean => {
  return (
    params.id === caller.id &&
    removeDecoratorsFromPhoneNumber(caller.primaryPhoneNumber) ===
      removeDecoratorsFromPhoneNumber(params.phoneNumber)
  )
}
