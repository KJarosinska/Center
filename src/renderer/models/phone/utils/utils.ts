import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
