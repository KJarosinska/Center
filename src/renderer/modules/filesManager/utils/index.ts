import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
