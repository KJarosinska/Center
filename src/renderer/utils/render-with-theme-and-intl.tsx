import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={LANGUAGE.default}
        locale={LANGUAGE.default}
        messages={localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}

export function renderWithTheme(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  const originalConsoleError = console.error
  console.error = (...args: any) => {
    if (
      !(
        args[0].startsWith("[React Intl] Cannot format message:") ||
        args[0].startsWith("[React Intl] Missing message:")
      )
    ) {
      originalConsoleError.call(console, ...args)
    }
  }
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider locale={LANGUAGE.default}>{ui}</IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
