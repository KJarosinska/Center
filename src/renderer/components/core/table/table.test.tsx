import "@testing-library/jest-dom/extend-expect"
import React from "react"
import {
  basicRows,
  nestedRows,
} from "Renderer/components/core/table/table.fake-data"
import {
  BasicTable,
  NestedTable,
} from "Renderer/components/core/table/table.stories"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"

const renderBasicTable = ({ sidebarOpened = false } = {}) => {
  const outcome = renderWithThemeAndIntl(
    <BasicTable sidebarOpened={sidebarOpened} />
  )
  return {
    ...outcome,
    getRows: () => outcome.queryAllByTestId("row"),
    getColumnsLabels: () => outcome.queryAllByTestId("column-label"),
  }
}

const renderNestedTable = ({ sidebarOpened = false } = {}) => {
  const outcome = renderWithThemeAndIntl(
    <NestedTable sidebarOpened={sidebarOpened} />
  )
  return {
    ...outcome,
    getRows: () => outcome.queryAllByTestId("row"),
    getNestedRows: () => outcome.queryAllByTestId("nested-row"),
    getColumnsLabels: () => outcome.queryAllByTestId("column-label"),
  }
}

test("basic table matches snapshot", () => {
  const { container } = renderBasicTable()
  expect(container).toMatchSnapshot()
})

test("basic table render rows properly", () => {
  const { getRows } = renderBasicTable()
  expect(getRows()).toHaveLength(basicRows.length)
})

test("basic table render columns labels properly", () => {
  const { getColumnsLabels } = renderBasicTable()
  expect(getColumnsLabels()).toHaveLength(2)
  expect(getColumnsLabels()[0]).toHaveTextContent("Name")
  expect(getColumnsLabels()[1]).toHaveTextContent("Phone")
})

test("basic table render columns data properly", () => {
  const { getRows } = renderBasicTable()
  const cols = getRows()[5].querySelectorAll("div")
  const rowData = basicRows[5]

  expect(cols[0]).toHaveTextContent(`${rowData.firstName} ${rowData.lastName}`)
  expect(cols[1]).toHaveTextContent(rowData.phoneNumber)
})

test("basic table with sidebar opened render columns properly", () => {
  const { getRows, getColumnsLabels } = renderBasicTable({
    sidebarOpened: true,
  })
  const cols = getRows()[5].querySelectorAll("div")
  expect(cols[0]).toBeVisible()
  expect(cols[1]).not.toBeVisible()

  expect(getColumnsLabels()[0]).toBeVisible()
  expect(getColumnsLabels()[1]).not.toBeVisible()
})

test("nested table matches snapshot", () => {
  const { container } = renderNestedTable()
  expect(container).toMatchSnapshot()
})

test("nested table render rows properly", () => {
  const { getRows, getNestedRows } = renderNestedTable()
  expect(getRows()).toHaveLength(nestedRows.length)
  expect(getNestedRows()).toHaveLength(3)
})

test("nested table render columns labels properly", () => {
  const { getColumnsLabels } = renderNestedTable()
  expect(getColumnsLabels()).toHaveLength(3)
  expect(getColumnsLabels()[0]).toHaveTextContent("File type")
  expect(getColumnsLabels()[1]).toHaveTextContent("Last backup")
  expect(getColumnsLabels()[2]).toHaveTextContent("Size")
})

test("nested table render columns data properly", () => {
  const { getNestedRows } = renderNestedTable()
  const cols = getNestedRows()[2].querySelectorAll("div")
  const rowData = nestedRows[2]._children?.[2]

  expect(rowData).toBeTruthy()

  if (rowData) {
    expect(cols[0]).toHaveTextContent(rowData.fileType)
    expect(cols[1]).toHaveTextContent(
      new Date(rowData.lastBackup).toLocaleString()
    )
    expect(cols[2]).toHaveTextContent(rowData.size)
  }
})

test("nested table with sidebar opened render columns properly", () => {
  const { getRows, getColumnsLabels } = renderNestedTable({
    sidebarOpened: true,
  })
  const cols = getRows()[2].querySelectorAll("div")

  expect(cols[0]).toBeVisible()
  expect(cols[1]).toBeVisible()
  expect(cols[2]).not.toBeVisible()

  expect(getColumnsLabels()[0]).toBeVisible()
  expect(getColumnsLabels()[1]).toBeVisible()
  expect(getColumnsLabels()[2]).not.toBeVisible()
})
