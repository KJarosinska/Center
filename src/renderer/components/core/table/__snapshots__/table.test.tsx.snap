// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic table matches snapshot 1`] = `
<div>
  <div
    class="sc-bZQynM QJGKs sc-gzVnrw jikWmX"
  >
    <div
      class="sc-bxivhb sc-EHOje fYapXe"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Name
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        Phone
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Reanna
         
        Shields
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        676-898-0415 x43082
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Russel
         
        Bahringer
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        840-783-7369 x379
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Dandre
         
        Hartmann
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        301.518.0527 x62072
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Sammy
         
        Simonis
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-073-230-8404 x6618
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Melisa
         
        Walter
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (201) 929-7783 x563
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Gaylord
         
        Boyer
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (787) 017-3647 x311
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Cassandra
         
        Vandervort
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        575.243.4091 x53650
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Sadie
         
        Tillman
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        332.411.9871 x80468
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Leola
         
        Nitzsche
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        345.038.1647
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Aaron
         
        Jacobson
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-556-376-4221 x9649
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Ethelyn
         
        Gutkowski
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-440-908-2294
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Jamey
         
        Koch
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        637-120-5137
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Gregorio
         
        Kulas
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (470) 212-1565 x5735
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Marjory
         
        Krajcik
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        133-287-4458
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Blaise
         
        Collins
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (430) 891-4246
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Odell
         
        Kreiger
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-997-582-4472
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Allene
         
        Medhurst
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (661) 745-0814 x2162
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Celia
         
        Kunde
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        399.818.3337
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Maximilian
         
        Toy
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (217) 479-0020
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Rosalinda
         
        Luettgen
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-227-354-4087 x717
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Justice
         
        Davis
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-137-214-5289 x56725
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Jaunita
         
        Little
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (284) 364-9520
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Bessie
         
        Graham
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-431-576-8718
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Arnulfo
         
        Schneider
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        776.195.3639
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Chase
         
        Mante
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (491) 629-4356
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Korey
         
        Raynor
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (493) 994-5418 x33523
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Patience
         
        Smitham
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-963-004-6333
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Maddison
         
        Kohler
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (417) 154-6086 x91300
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Ray
         
        Pfeffer
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        109.435.9973
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Adelia
         
        Upton
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        1-745-123-5568 x824
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Alda
         
        Kuphal
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        (768) 575-8988 x41000
      </div>
    </div>
    <div
      class="sc-bxivhb jxsmHN"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH jMtnXt"
      >
        Brady
         
        Koepp
      </div>
      <div
        class="sc-bwzfXH sc-htpNat jGhPcP"
      >
        791.853.3917
      </div>
    </div>
  </div>
</div>
`;

exports[`nested table matches snapshot 1`] = `
<div>
  <div
    class="sc-EHOje gRHaHa sc-bZQynM cRCinu"
  >
    <div
      class="sc-htpNat sc-ifAKCX ehYSDj"
    >
      <div
        class="sc-bwzfXH uTxCW"
        data-testid="column-label"
      >
        File type
      </div>
      <div
        class="sc-bwzfXH uTxCW"
        data-testid="column-label"
      >
        Last backup
      </div>
      <div
        class="sc-bwzfXH uTxCW"
        data-testid="column-label"
      >
        Size
      </div>
    </div>
    <div
      class="sc-htpNat dFjmtx"
      data-testid="row"
      size="4.8"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Messages
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        8/22/2019, 4:10:30 AM
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        10.31 MB
      </div>
    </div>
    <div
      class="sc-htpNat dFjmtx"
      data-testid="row"
      size="4.8"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Contacts
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        3/13/2019, 2:34:01 AM
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        15.99 MB
      </div>
    </div>
    <div
      class="sc-htpNat dFjmtx"
      data-testid="row"
      size="4.8"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Files
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        3/19/2019, 5:23:13 AM
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        12.83 MB
      </div>
    </div>
    <div
      class="sc-bxivhb diDWuo"
    >
      <div
        class="sc-htpNat gSmDhk"
        data-testid="nested-row"
        size="4"
      >
        <div
          class="sc-bwzfXH uTxCW"
        >
          Music files
        </div>
        <div
          class="sc-bwzfXH uTxCW"
        >
          3/8/2019, 1:33:03 PM
        </div>
        <div
          class="sc-bwzfXH uTxCW"
        >
          12.33 MB
        </div>
      </div>
      <div
        class="sc-htpNat gSmDhk"
        data-testid="nested-row"
        size="4"
      >
        <div
          class="sc-bwzfXH uTxCW"
        >
          Recorded files
        </div>
        <div
          class="sc-bwzfXH uTxCW"
        >
          11/27/2019, 8:29:41 AM
        </div>
        <div
          class="sc-bwzfXH uTxCW"
        >
          31.68 MB
        </div>
      </div>
      <div
        class="sc-htpNat gSmDhk"
        data-testid="nested-row"
        size="4"
      >
        <div
          class="sc-bwzfXH uTxCW"
        >
          Storage files
        </div>
        <div
          class="sc-bwzfXH uTxCW"
        >
          4/17/2019, 4:42:38 PM
        </div>
        <div
          class="sc-bwzfXH uTxCW"
        >
          4.92 MB
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat dFjmtx"
      data-testid="row"
      size="4.8"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Notes
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        10/8/2019, 7:51:33 PM
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        24.36 MB
      </div>
    </div>
    <div
      class="sc-htpNat dFjmtx"
      data-testid="row"
      size="4.8"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Meditation timer data
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        10/13/2019, 3:15:12 AM
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        44.32 MB
      </div>
    </div>
  </div>
</div>
`;

exports[`simple table matches snapshot 1`] = `
<div>
  <div
    class="sc-EHOje gRHaHa"
  >
    <div
      class="sc-htpNat sc-ifAKCX ehYSDj"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Name
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        Phone
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Kiley
         
        Collier
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        451-981-6719 x781
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Lambert
         
        Nicolas
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        655-667-8937 x2052
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Euna
         
        Lebsack
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-910-757-7424 x53888
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Muriel
         
        Wunsch
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        946-568-8407 x8728
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Geraldine
         
        Marks
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-579-872-1729
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Ettie
         
        Rath
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-799-456-8290
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Rahul
         
        Powlowski
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-163-737-1009 x389
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Efren
         
        Watsica
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        548.041.1643
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Mariah
         
        Cummerata
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-785-636-4270
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Gerry
         
        Toy
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        651.722.8112 x27851
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Ettie
         
        Lesch
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        (787) 119-1261 x58468
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Pearlie
         
        Torphy
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        318.476.1845 x631
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Jermain
         
        Crist
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        634-227-9124
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Winfield
         
        Tromp
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        457-514-3661 x5418
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Verda
         
        Thompson
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-935-995-3325 x36257
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Jerad
         
        Graham
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        (481) 429-6193 x4315
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Gerhard
         
        Rempel
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        126.059.4195 x787
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Roel
         
        Kuhic
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        162-378-4754 x2769
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Wilburn
         
        Prosacco
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        039-060-8196 x16140
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Izabella
         
        Schmitt
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-866-877-0732
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Kaleigh
         
        Kessler
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        547.117.9907 x56741
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Heber
         
        Mohr
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-354-068-7649
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Audie
         
        Ruecker
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        902-440-8097
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Marvin
         
        Schowalter
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        (599) 201-9530
      </div>
    </div>
    <div
      class="sc-htpNat iKPCwm"
      data-testid="row"
    >
      <div
        class="sc-bwzfXH uTxCW"
      >
        Ethelyn
         
        Hagenes
      </div>
      <div
        class="sc-bwzfXH uTxCW"
      >
        1-065-466-9042 x0206
      </div>
    </div>
  </div>
</div>
`;
